modin/core/execution/ray/common/utils.py

```diff
@@ -246,3 +246,26 @@ def deserialize(obj):
         return dict(zip(obj.keys(), RayWrapper.materialize(list(obj.values()))))
     else:
         return obj
+
+
+def wait(obj_ids):
+    """
+    Wrap ``ray.wait`` to handle duplicate object references.
+
+    ``ray.wait`` assumes a list of unique object references: see
+    https://github.com/modin-project/modin/issues/5045
+
+    Parameters
+    ----------
+    obj_ids : List[ObjectIDType]
+        The object IDs to wait on.
+
+    Returns
+    -------
+    Tuple[List[ObjectIDType], List[ObjectIDType]]
```

Comment:
Don't we return only one object here, i.e. just a list?

Reply:
I don't think so. See here: https://docs.ray.io/en/latest/_modules/ray/_private/worker.html#wait

Reply:
I believe if we specify the `num_returns` parameter, then we wait until all of the passed ObjectRefs are available in the object store.

cc @modin-project/modin-ray

Reply:
Yes, I think in this case since we're waiting on every single object, all of them would be completed and this would return `([list of objects], ())`. I kept the function like this so it would match with the behavior of `ray.wait`, though we don't actually use the return value of the function anywhere.